engine/source/public/core/asset.hpp

```diff
@@ -0,0 +1,70 @@
+﻿#pragma once
+#include <string>
+
+namespace lumina
+{
+    enum class AssetType
+    {
+        Texture, // Including Samplers or not can be discussed
+        Shader,
+        Model, // Collections of meshes
+        Mesh,
+        Material,
+        None
+    };
+
+    class Asset
+    {
+        friend class AssetManager;
+
+    private:
+        static size_t nextID;
+
+    protected:
+        size_t id;
+        AssetType type;
+        std::string filePath;
```

Comment:
do we want to initialize these with initial values or with `{}`?

Reply:
I did this before, but id and type get initialized in the constructor, std::string filePath can be initialized to {}